File: hackerrank/counting-sort/counting-sort-2/solution.ts
function countingSort(arr: number[]): number[] {
  const freq = new Array(100).fill(0);

  for (const num of arr) {
    freq[num]++;
  }

  let idx: number = 0;
  for (let i = 0; i < 100; i++) {
    for (let j = 0; j < freq[i]; j++) {
      arr[idx++] = i;
    }
  }

  return arr;
}

File: hackerrank/counting-sort/counting-sort-2/solution.cpp
vector<int> countingSort(vector<int> arr) {
    vector<int> freq(100, 0);

    for (int i = 0; i < arr.size(); i++) {
        freq[arr[i]]++;
    }

    int idx = 0;
    for (int i = 0; i < 100; i++) {
        for (int j = 0; j < freq[i]; j++) {
            arr[idx++] = i;
        }
    }

    return arr;
}
